design/authorization-handler.md

```diff
@@ -0,0 +1,601 @@
+
+```json
+{
+	"type": "http://keycloak.org/auth-type/group",
+	"identifier": "123"
+}
+```
+
+Given an Authorization Handler with a ScopeParameterParser configured with a regexp such as `tenant:(.*)#group:(.*)`. Scope parameter value like `tenant:123#group:456` would correspond to JSON like this:
+
+```json
+{
+	"type": "http://keycloak.org/auth-type/tenant-group",
+	"identifier": {
+    "tenant": "123",
+    "group": "456"
+  }
+}
+```
+
+### StaticScopeParser
+
+This parser would work as the default ClientScopes in Keycloak. An Authorization Handler would be created and defined as a Static Scope parser with the scope that will be supported. 
+
+A static OAuth scope such as `address` will be parsed similar to the previous cases, but with the `identifier` field missing:
+```json
+{
+	"type": "address"
+}
+```
+
+The client scope will be determined again from the type field.
+
+The request will be rejected in case that there are any unknown scope values or unknown authorization_detail types, which don't correspond to any configured authorization handlers in the realm. The error returned to the client is `invalid_authorization_details` as described in the [RAR specification Section 8](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-07#section-8) or `invalid_scope` as it works today and described in the OAuth2 specification.
+
+### ClientAuthorizationContext in AuthenticationSessionModel
+
+The ClientAuthorizationContext can be sent as a field in the AuthenticationSessionModel. Hence it will be available in all the requests of this Authentication session. Existing AuthenticationSession methods “get/setClientScopes” may be deprecated and replaced by “get/setAuthenticationContext”
+
+## Compliance processor
+
+There will be a chain of policies configured in the admin console for a specific Authorization  Handler. The ClientAuthorizationContext can be processed by this policy chain. Each AuthorizationDetailsEntry will be processed by each of the policy and policy will decide if:
+
+* AuthorizationDetails is required to be rendered on the consent screen
+* AuthorizationDetails is unsupported/invalid/unauthorized and whole client request should be rejected
+
+Policy can also “enrich” and/or change authorizationDetails entries as sometimes client sent “incomplete” authorization_details and it is up to the AS (Keycloak server) to figure correct values and then in later stages, display them to the user on consent screen or add them to the token. Some use-case described in the [RAR specification Section 7.1](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-07#section-7.1).
+
+Policy is also responsible for processing the consent screen (The moment when a user submits the consent screen, which can happen multiple times during a single “authentication session” due the fact that more consent screens would be possible with this proposal).
+
+### Naming
```

Comment:
Minor: For some reason, this is not shown as "Header" when I try to view this text as HTML. See https://github.com/keycloak/keycloak-community/blob/1f6b5f86c40eb7688069ee39ef3e2c8d4fd8d81d/design/authorization-handler.md . It may be only github bug, not sure. This is also only minor cosmetic thing.

Same issue can be seen for multiple places (EG. for "When" section below as well).

Reply:
Yeah it happened on quite a few headers in the document for some reason. Fixing them all.